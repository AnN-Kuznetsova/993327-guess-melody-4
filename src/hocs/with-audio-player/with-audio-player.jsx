import React, {PureComponent} from 'react';
import {AudioPlayer} from "../../components/audio-player/audio-player.jsx";

export const withAudioPlayer = (Component) => {
  class WithAudioPlayer extends PureComponent {
    constructor(props) {
      super(props);

      this.state = {
        activePlayerId: 0,
      };
    }

    _handlePlayButtonClick(id) {
      const {activePlayerId} = this.state;

      this.setState({
        activePlayerId: activePlayerId === id ? -1 : id
      });
    }

    _renderPlayer(src, id) {
      const {activePlayerId} = this.state;

      return (
        <AudioPlayer
          src={src}
          isPlaying={id === activePlayerId}
          onPlayButtonClick={this._handlePlayButtonClick.bind(this, id)}
        />
      );

    }

    render() {
      return <Component
        {...this.props}
        renderPlayer={this._renderPlayer.bind(this)}
      />;
    }
  }


  WithAudioPlayer.propTypes = {};


  return WithAudioPlayer;
};
