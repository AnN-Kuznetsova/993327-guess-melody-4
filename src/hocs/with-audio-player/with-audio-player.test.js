import React from "react";
import renderer from "react-test-renderer";
import {withAudioPlayer} from "./with-audio-player.jsx";


const Component = (props) => {
  return (
    <div {...props} />
  );
};


describe(`Render withAudioPlayer`, () => {
  it(`Should match with snapshot`, () => {
    const withAudioPlayerSnapshot = renderer.create(
        <withAudioPlayer Component={Component} />
    ).toJSON();

    expect(withAudioPlayerSnapshot).toMatchSnapshot();
  });
});
