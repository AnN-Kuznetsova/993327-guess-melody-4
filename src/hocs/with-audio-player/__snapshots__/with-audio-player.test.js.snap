// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render withAudioPlayer Should match with snapshot 1`] = `
<withAudioPlayer
  Component={[Function]}
/>
`;
